src/scripts/main.js

```diff
@@ -1,3 +1,67 @@
 'use strict';
 
-// write code here
+const tBody = document.querySelector('tbody');
+const appRow = document.querySelector('.append-row');
+const remRow = document.querySelector('.remove-row');
+const appCol = document.querySelector('.append-column');
+const remCol = document.querySelector('.remove-column');
+
+appCol.addEventListener('click', (e) => {
+  const tabRows = document.querySelectorAll('tr');
+
+  for (const row of tabRows) {
+    const td = document.createElement('td');
+
+    row.appendChild(td);
+
+    if (row.childElementCount === 3) {
+      remCol.disabled = false;
+    }
+
+    if (row.childElementCount === 10) {
+      appCol.disabled = true;
+    }
+  }
+});
+
+remCol.addEventListener('click', (e) => {
+  const tabRows = document.querySelectorAll('tr');
+
+  for (const row of tabRows) {
+    row.removeChild(row.lastChild);
+
+    if (row.childElementCount === 2) {
+      remCol.disabled = true;
+    }
+
+    if (row.childElementCount === 9) {
+      appCol.disabled = false;
+    }
+  }
+});
```

Comment:
Add single eventListener to the entire table, use switch..case to check which button was clicked and run needed actions. After switch use such code to disable needed buttons
```javascript
remCol.disabled = row.childElementCount === 2;
```